src/agent/agent.js

```diff
@@ -260,7 +260,15 @@ export class Agent {
         this.bot.on('messagestr', async (message, _, jsonMsg) => {
             if (jsonMsg.translate && jsonMsg.translate.startsWith('death') && message.startsWith(this.name)) {
                 console.log('Agent died: ', message);
-                this.handleMessage('system', `You died with the final message: '${message}'. Previous actions were stopped and you have respawned. Notify the user and perform any necessary actions.`);
+                let death_pos = this.bot.entity.position;
+                this.memory_bank.rememberPlace('last death position', death_pos);
```

Comment:
can you save as "last_death_position" instead

Reply:
OH also you probably want save the location on the 'death' event, instead of here. The bot may have already respawned, and the location will be wrong. the death event should be triggered earlier

Reply:
oh that's why it kept breaking. smart. thankyou I'll fix that! 👍

Reply:
it turns out the message is actually received first (or its a race condition)
but its seems to be more consistent for the "messagestr" event

it looks like the "death" event is actually called after it respawned
because when I tried saving the death position in the "death" event `this.agent.memory_bank.recallPlace('last_death_position')[0]` would return null
I believe this is because the "messagestr" event is called first
I tried using "setTimeout()" with a small delay but it thought its spawn was its death position
(probably because "death" event is called after it respawned)

TLDR: saving the death position in the "messagestr" event doesn't work

Reply:
> can you save as "last_death_position" instead

also I did this and didn't mention it in my [modified message sent to AI] (https://github.com/kolbytn/mindcraft/pull/252/commits/eb0719d4e378ca72bfb04cb6d5d8a8f7e782389b) commit